email_handler.py

```diff
@@ -518,11 +518,19 @@ def handle_reply(envelope, smtp: SMTP, msg: Message, rcpt_to: str) -> (bool, str
     delete_header(msg, "Received")
 
     # make the email comes from alias
+    from_header = alias.email
+    # add alias name from alias
     if alias.name:
         LOG.d("Put alias name in from header")
         from_header = formataddr((alias.name, alias.email))
-    else:
-        from_header = alias.email
+    elif alias_domain not in ALIAS_DOMAINS:
```

Comment:
We can use `alias.custom_domain_id` here I think.  This field is filled up if the alias is dynamically created thanks to the directory feature.

We can also create a relationship like `custom_domain = db.relationship("CustomDomain", foreign_keys=[custom_domain_id])` to avoid doing a `CustomDomain.get()`, this is the same in terms of number of query to the database though.

Reply:
I ended up adding the relationship in the model, this enabled me to display the domain-wide alias name as a placeholder in the alias view.

This does make alias.custom_domain_id redundant, but it has quite a few users left.